core-bundle/tests/Command/UserListCommandTest.php

```diff
@@ -42,84 +43,164 @@ public function testDefinition(): void
         $this->assertTrue($definition->hasOption('admins'));
     }
 
-    public function testTakesAdminsFlagAsArgument(): void
+    public function testThrowsExceptionOnInvalidFormat(): void
     {
-        $command = $this->getCommand();
+        $this->expectException(\LogicException::class);
 
-        $input = [
-            '--admins' => true,
-        ];
-
-        $code = (new CommandTester($command))->execute($input);
+        $command = $this->getCommand($this->mockQueryBuilder([]));
 
-        $this->assertSame(0, $code);
+        $this->executeCommand($command, ['--format' => 'foo'], 1);
     }
 
-    public function testReturnsValidJson(): void
+    /**
+     * @dataProvider listsUsersProvider
+     */
+    public function testListsUsers(array $input, array $data, string $expected): void
     {
-        $command = $this->getCommand();
+        $command = $this->getCommand($this->mockQueryBuilder($data));
 
-        $input = [
-            '--format' => 'json',
-        ];
+        $output = $this->executeCommand($command, $input);
 
-        $commandTester = new CommandTester($command);
+        $this->assertStringContainsString($expected, $output);
+    }
 
-        $code = $commandTester->execute($input);
-        $output = $commandTester->getDisplay();
+    /**
+     * @dataProvider listsUsersProvider
+     */
+    public function testListsUsersAsJson(array $input, array $data, string $expectedTxt, array $expected): void
+    {
+        $command = $this->getCommand($this->mockQueryBuilder($data));
+        $input['--format'] = 'json';
+
+        $output = json_decode($this->executeCommand($command, $input), true);
 
-        $this->assertSame(0, $code);
-        $this->assertNotNull(json_decode($output, true));
+        $this->assertSame($expected, $output);
     }
 
-    public function testReturnsValidJsonWithSubset(): void
+    public function listsUsersProvider(): \Generator
     {
-        $command = $this->getCommand();
+        yield 'Returns empty result' => [
+            [],
+            [],
+            'No accounts found.',
+            [],
+        ];
 
-        $input = [
-            '--format' => 'json',
-            '--column' => ['name', 'username'],
+        yield 'Returns empty result with column argument' => [
+            ['--column' => ['firstname', 'lastname']],
+            [],
+            'No accounts found.',
+            [],
         ];
 
-        $commandTester = new CommandTester($command);
+        yield 'Returns default fields from data' => [
+            [],
+            [['id' => 42, 'username' => 'k.jones', 'name' => 'Kevin Jones', 'admin' => '1', 'dateAdded' => time(), 'lastLogin' => time()]],
+            'k.jones    Kevin Jones',
+            [['username' => 'k.jones', 'name' => 'Kevin Jones', 'admin' => '1', 'dateAdded' => time(), 'lastLogin' => time()]],
```

Comment:
```suggestion
            [['id' => 42, 'username' => 'k.jones', 'name' => 'Kevin Jones', 'admin' => '1', 'dateAdded' => 1697459597, 'lastLogin' => 1697459597]],
            'k.jones    Kevin Jones',
            [['username' => 'k.jones', 'name' => 'Kevin Jones', 'admin' => '1', 'dateAdded' => 1697459597, 'lastLogin' => 1697459597]],
```

Using `time()` would cause issues as sometimes the later call would return a higher value.